addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -999,13 +1123,66 @@ var SnippetEditor = Widget.extend({
+
+            // Defining the column grid area with its position.
+            const gridProp = gridUtils.getGridProperties(rowEl);
+
+            const top = parseFloat(this.$target[0].style.top);
+            const left = parseFloat(this.$target[0].style.left);
+
+            const rowStart = Math.round(top / (gridProp.rowSize + gridProp.rowGap)) + 1;
+            const columnStart = Math.round(left / (gridProp.columnSize + gridProp.columnGap)) + 1;
+            const rowEnd = rowStart + this.dragState.columnRowCount;
+            const columnEnd = columnStart + this.dragState.columnColCount;
+
+            this.$target[0].style.gridArea = `${rowStart} / ${columnStart} / ${rowEnd} / ${columnEnd}`;
+
+            // Cleaning.
+            gridUtils.gridCleanUp(rowEl, this.$target[0]);
+
+            // Setting the z-index to the maximum of the grid.
+            gridUtils.setColumnToMaxZindex(this.$target[0], rowEl);
+
+        } else if (this.$target[0].classList.contains('o_grid_item') && this.dropped) {
+            // Case when dropping a grid item in a non-grid dropzone.
+            this.$target[0].classList.remove('o_grid_item');
+            this.$target[0].style.removeProperty('grid-area');
+            this.$target[0].style.removeProperty('z-index');
+        }
+
         // TODO lot of this is duplicated code of the d&d feature of snippets
         if (!this.dropped) {
             const { nearest } = this.$body[0].ownerDocument.defaultView.$;
             let $el = nearest({x: ui.position.left, y: ui.position.top}, '.oe_drop_zone', {container: document.body}).first();
             // Some drop zones might have been disabled.
             $el = $el.filter(this.$dropZones);
             if ($el.length) {
+                // If the column is not dropped inside a dropzone.
+                if (this.$target[0].classList.contains('o_grid_item')) {
+                    if ($el[0].classList.contains('oe_grid_zone')) {
+                        // Case when a column is dropped near a grid.
+                        // Placing it in the top left corner.
+                        this.$target[0].style.gridArea = `1 / 1 / ${1 + this.dragState.columnRowCount} / ${1 + this.dragState.columnColCount}`;
+                        const rowEl = $el[0].parentNode;
+                        const rowCount = Math.max(rowEl.dataset.rowCount, 1 + this.dragState.columnRowCount);
+                        rowEl.dataset.rowCount = rowCount;
+
+                        // Setting the z-index to the maximum of the grid.
+                        gridUtils.setColumnToMaxZindex(this.$target[0], rowEl);
+
+                    } else {
+                        // Case when a column is dropped near a non-grid dropzone.
+                        this.$target[0].classList.remove('o_grid_item');
+                        this.$target[0].style.removeProperty('z-index');
+                    }
+                }
```

Comment:
Just wondering: can't this logic be done in the related snippet option's `onMove` method ? Basically evey code that can be in the snippet option instead of the generic behavior should be in that snippet option... but maybe this is not possible here, tell me.

Reply:
It is not possible to do that here because the call to `onMove` is made after the destruction of all the dropzones and this part depends on them, as it finds the nearest one to put the element there.